packages/neon-core/src/sc/manifest/ContractManifest.ts

```diff
@@ -0,0 +1,107 @@
+import { ContractGroup, ContractGroupLike } from "./ContractGroup";
+import { ContractFeatures, ContractFeaturesLike } from "./ContractFeatures";
+import { ContractAbi, ContractAbiLike } from "./ContractAbi";
+import {
+  ContractPermission,
+  ContractPermissionLike
+} from "./ContractPermission";
+import { WildCardContainer } from "./WildCardContainer";
+import { ContractMethodDescriptor } from "./ContractMethodDescriptor";
+import { ContractEventDescriptor } from "./ContractEventDescriptor";
+
+export interface ContractManifestLike {
+  groups: ContractGroupLike[];
+  features: ContractFeaturesLike;
+  abi: ContractAbiLike;
+  permissions: ContractPermissionLike[];
+  trusts: "*" | string[];
+  safeMethods: "*" | string[];
+}
+
+export class ContractManifest {
+  public static readonly MAX_LENGTH = 2048;
+  public hash: string;
+  public groups: ContractGroup[];
+  public features: ContractFeatures;
+  public abi: ContractAbi;
+  public permissions: ContractPermission[];
+  public trusts: WildCardContainer;
+  public safeMethods: WildCardContainer;
+
+  public constructor(obj: Partial<ContractManifestLike>) {
+    const {
+      groups = [],
+      features: { payable, storage } = { payable: false, storage: false },
+      abi = {},
+      permissions = [],
+      trusts = "*",
+      safeMethods = "*"
+    } = obj;
+    this.groups = groups.map(group => new ContractGroup(group));
+    this.features = new ContractFeatures(storage, payable);
+    this.abi = new ContractAbi(abi);
+    this.permissions = permissions.map(
+      permission => new ContractPermission(permission)
+    );
+    this.trusts = WildCardContainer.fromSerialized(trusts);
+    this.safeMethods = WildCardContainer.fromSerialized(safeMethods);
+    this.hash = this.abi.hash;
+  }
+
+  public static createDefault(hash: string): ContractManifest {
```

Comment:
do we need a createDefault if we can just use the constructor with empty args to create it? Is there anything special with this default?

Reply:
Removed all the defaults, cause I don't see now how we can use them.